UTY-2275 buildsystem refactor (#1228)
spatialos/gdk-for-unity: CHANGELOG.md

## Unreleased

> ### Added

Reviewer: Did the semantics of the build order change in this PR? I recall you talking about this

Author: No

> - Added support for Unity build callbacks such as `IPostprocessBuildWithReport`.

Reviewer: ```suggestion
- Added support for Unity build callbacks such as `IPostprocessBuildWithReport`. [#1228](https://github.com/spatialos/gdk-for-unity/pull/1228)
```

>   - During a build you will have access to the `WorkerBuilder.CurrentContext` field which contains all SpatialOS related build information.

Reviewer 2: maybe?
```suggestion
  - During a build you can now access the `WorkerBuilder.CurrentContext` field which contains all SpatialOS related build information.
```
